tests/sentry/workflow_engine/processors/test_data_condition_group.py

```diff
@@ -0,0 +1,253 @@
+    def test_process_data_condition_group(self):
+        with mock.patch(
+            "sentry.workflow_engine.processors.data_condition_group.logger"
+        ) as mock_logger:
+            assert process_data_condition_group(1, 1) == (False, [])
+            assert mock_logger.exception.call_args[0][0] == "DataConditionGroup does not exist"
+
+    def test_process_data_condition_group__exists__fails(self):
+        data_condition_group = self.create_data_condition_group()
+        self.create_data_condition(
+            condition_group=data_condition_group, condition="gt", comparison="5"
+        )
+
+        assert process_data_condition_group(data_condition_group.id, 1) == (False, [])
+
+    def test_process_data_condition_group__exists__passes(self):
+        data_condition_group = self.create_data_condition_group()
+        self.create_data_condition(
+            condition_group=data_condition_group,
+            condition="gt",
+            comparison="5",
+            condition_result=DetectorPriorityLevel.HIGH,
+        )
+        assert process_data_condition_group(data_condition_group.id, 10) == (
+            True,
+            [DetectorPriorityLevel.HIGH],
+        )
+
+
+class TestEvaluateConditionGroupTypeAny(TestCase):
+    def setUp(self):
+        self.data_condition_group = self.create_data_condition_group(
+            logic_type=DataConditionGroup.Type.ANY
+        )
+
+        self.data_condition = self.create_data_condition(
+            condition="gt",
+            comparison="5",
+            condition_result=DetectorPriorityLevel.HIGH,
+            condition_group=self.data_condition_group,
+        )
+
+        self.data_condition_two = self.create_data_condition(
+            condition="gt",
+            comparison="3",
+            condition_result=DetectorPriorityLevel.LOW,
+            condition_group=self.data_condition_group,
+        )
+
+        self.conditions = [self.data_condition, self.data_condition_two]
+
+    def test_evaluate_condition_group__passes_all(self):
+        assert evaluate_condition_group(
+            self.data_condition_group,
+            10,
+        ) == (
+            True,
+            [DetectorPriorityLevel.HIGH, DetectorPriorityLevel.LOW],
+        )
+
+    def test_evaluate_condition_group__passes_one(self):
+        assert evaluate_condition_group(
+            self.data_condition_group,
+            4,
+        ) == (
+            True,
+            [DetectorPriorityLevel.LOW],
+        )
+
+    def test_evaluate_condition_group__fails_all(self):
+        assert evaluate_condition_group(
+            self.data_condition_group,
+            1,
+        ) == (
+            False,
+            [],
+        )
+
+    def test_evaluate_conditon_group__passes_without_conditions(self):
+        data_condition_group = self.create_data_condition_group(
+            logic_type=DataConditionGroup.Type.ANY
+        )
+        assert evaluate_condition_group(data_condition_group, 10) == (
```

Comment:
Should it be possible to have no conditions? Also there's a typo in the test name `conditon`